addons/knowledge/models/knowledge_article.py

```diff
@@ -358,18 +399,20 @@ def filter_on_permission(members, permission):
 
     def _compute_is_user_favourite(self):
         for article in self:
-            article.is_user_favourite = self.env.user in article.favourite_user_ids
+            article.is_user_favourite = self.env.user in article.favourite_user_ids.mapped('user_id')
```

Comment:
this will need a sudo I think -> To allow base users to read the model of favourite_user_ids

Reply:
I changed the permissions so that users and portal can do whatever they want (read/create/write/delete) `knowledge.article.favourite` records with their user_id and if they have access to the `article_id`